src/m365/onenote/commands/notebook/notebook-add.spec.ts

```diff
@@ -0,0 +1,633 @@
+      if (opts.url === `https://graph.microsoft.com/v1.0/me/onenote/notebooks`) {
+        return Promise.resolve({
+          "id": "1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0",
+          "self": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0",
+          "createdDateTime": "2022-10-26T00:05:46Z",
+          "displayName": "Private Note",
+          "lastModifiedDateTime": "2022-10-26T00:05:46Z",
+          "isDefault": false,
+          "userRole": "Owner",
+          "isShared": false,
+          "sectionsUrl": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0/sections",
+          "sectionGroupsUrl": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0/sectionGroups",
+          "createdBy": {
+            "user": {
+              "id": "am917f88-cd36-4048-83c7-6z6608f344f0",
+              "displayName": "John Doe"
+            }
+          },
+          "lastModifiedBy": {
+            "user": {
+              "id": "am917f88-cd36-4048-83c7-6z6608f344f0",
+              "displayName": "John Doe"
+            }
+          },
+          "links": {
+            "oneNoteClientUrl": {
+              "href": "onenote:https://contoso-my.sharepoint.com/personal/jdoe_contoso_onmicrosoft_com/Documents/Notebooks/Private%20Notebook"
+            },
+            "oneNoteWebUrl": {
+              "href": "https://contoso-my.sharepoint.com/personal/jdoe_contoso_onmicrosoft_com/Documents/Notebooks/Private%20Notebook"
+            }
+          }
+        });
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    await command.action(logger, { options: { name: 'Private Notebook', debug: true } });
+    assert(loggerLogSpy.calledWith({
+      "id": "1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0",
+      "self": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0",
+      "createdDateTime": "2022-10-26T00:05:46Z",
+      "displayName": "Private Note",
+      "lastModifiedDateTime": "2022-10-26T00:05:46Z",
+      "isDefault": false,
+      "userRole": "Owner",
+      "isShared": false,
+      "sectionsUrl": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0/sections",
+      "sectionGroupsUrl": "https://graph.microsoft.com/v1.0/users/am917f88-cd36-4048-83c7-6z6608f344f0/onenote/notebooks/1-558ac4dh-3c0a-4123-bc46-e4d1c22256f0/sectionGroups",
+      "createdBy": {
+        "user": {
+          "id": "am917f88-cd36-4048-83c7-6z6608f344f0",
+          "displayName": "John Doe"
+        }
+      },
+      "lastModifiedBy": {
+        "user": {
+          "id": "am917f88-cd36-4048-83c7-6z6608f344f0",
```

Comment:
many times we are repeating the same mocked callback. Instead we could save it once to a variable at the top like `mockedResponse` and reuse the variable